crates/sui-types/src/temporary_store.rs

```diff
@@ -176,6 +176,36 @@ impl<S> TemporaryStore<S> {
         }
     }
 
+    /// WARNING! Should only be used for dry run and dev inspect!
+    /// In dry run and dev inspect, you might load a dynamic field that is actually too new for
+    /// the transaction. Ideally, we would want to load the "correct" dynamic fields, but as that
+    /// is not easily determined, we instead set the lamport version MAX, which is a valid lamport
+    /// version for any object used in the transaction (preventing internal assertions or
+    /// invariant violations from being triggered)
+    pub fn new_for_mock_transaction(
```

Comment:
I had to change this to not break things like balance changes, so thanks folks for the tests!

Reply:
This approach seems better, to be honest -- less invasive.